packages/widget-chat/src/containers/notifications/index.js

```diff
@@ -0,0 +1,125 @@
+import {Component} from 'react';
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+import _ from 'lodash';
+
+import {getUnsentNotifications} from '../../selectors/notifications';
+import {notificationSent, setNotificationPermission, setNotificationSupported} from '../../actions/notifications';
+
+const TIMEOUT_LENGTH = 10000;
+
+class Notifications extends Component {
+  shouldComponentUpdate(nextProps) {
+    const {props} = this;
+    return nextProps.notifications !== props.notifications || nextProps.isSupported !== props.isSupported || nextProps.permission !== props.permission;
+  }
+
+  componentDidUpdate() {
+    const {props} = this;
+    // Default state is not supported, once we know it is, don't check any more
+    if (!props.isSupported) {
+      this._checkSupported();
+    }
+    if (_.isNull(props.permission) && props.isSupported) {
+      Notification.requestPermission((permission) => props.setNotificationPermission(permission));
+    }
+    this._displayNotifications();
+  }
+
+  /**
+   * Checks if browser notifications are supported and updates store
+   *
+   * @returns {null}
+   */
+  _checkSupported() {
```

Comment:
I noticed you started prefixing underscores for some methods. Is this something we want to start doing? What are your rules for when to use the underscore? From this component it looks like methods that aren't a part of the component lifecycle.